src/sql/platform/connection/common/connectionProfileGroup.ts

```diff
@@ -157,6 +157,39 @@ export class ConnectionProfileGroup implements IConnectionProfileGroup {
 		});
 	}
 
+	public removeConnections(connections: ConnectionProfile[]): void {
+		connections.forEach((conn) => {
+			this._childConnections = this._childConnections.filter((curConn) => { return curConn.id !== conn.id; });
```

Comment:
Instead of filtering multiple times and reassigning it each time I'd switch it and loop through child connections once and have the filter check if the ID is in the `connections` list. We're much more likely to have more children than connections to remove.